pkg/controllers/counter/controller.go

```diff
@@ -102,11 +92,10 @@ func (c *Controller) resourceCountsFor(provisionerName string) v1.ResourceList {
 	return result
 }
 
-func (c *Controller) Builder(_ context.Context, m manager.Manager) operatorcontroller.Builder {
-	return controllerruntime.
+func (c *Controller) Builder(_ context.Context, m manager.Manager) corecontroller.TypedBuilder {
+	return corecontroller.NewTypedBuilderControllerRuntimeAdapter(controllerruntime.
```

Comment:
I'm not sold on this. This doesn't work as well with languages that don't support polymorphism, primarily because only one thing in the `controller` package can have the function signature named `Adapt`, so making the signature specific makes more sense.